dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -32,22 +32,27 @@ public DockerfileGitHubUtil(GitHubUtil gitHubUtil) {
 
     protected GitHubUtil getGitHubUtil() { return gitHubUtil; }
 
-    public GHRepository checkFromParentAndFork(GHRepository parent) throws IOException {
-        gitHubUtil.createFork(parent);
+    public GHRepository closeOutdatedPullRequestAndForkParent(GHRepository parent) throws IOException {
 
         for (GHRepository fork : parent.listForks()) {
             String forkOwner = fork.getOwnerName();
             GHMyself myself = gitHubUtil.getMyself();
             String myselfLogin = myself.getLogin();
             if (forkOwner.equals(myselfLogin)) {
-                if (!pullRequestAlreadyExists(parent)) {
-                    gitHubUtil.safeDeleteRepo(fork);
-                    return gitHubUtil.createFork(parent);
+                GHPullRequest pr = getPullRequestWithPullReqIdentifier(parent);
+                log.info("pr: {}", pr);
+                // Only reason we close the existing PR, delete fork and re-fork, is because there is no way to
+                // determine if the existing fork is compatible with it's parent.
+                if (pr != null) {
+                    // close the pull-request since the fork is out of date
+                    log.info("closing existing pr: {}", pr.getUrl());
+                    pr.close();
```

Comment:
Would it be better to close this in a try-finally?

Reply:
Good point. I think I can use try catch to log about not able to close PR.